packages/studio/src/hooks/useActiveComponentName.tsx

```diff
@@ -16,6 +17,27 @@ export default function useActiveComponentName(): string | undefined {
   return getComponentDisplayName(activeState);
 }
 
+export function useComponentNames(
+  uuids: Set<string>
+): { name: string; uuid: string }[] {
+  const componentTree = useStudioStore((store) => {
+    return store.actions.getComponentTree();
+  });
+  if (!componentTree) {
+    return [];
+  }
+  const componentStates = ComponentTreeHelpers.mapComponentTreeParentsFirst(
+    componentTree,
+    (c) => {
+      if (uuids.has(c.uuid)) return c;
+      else return undefined;
+    }
+  );
+  return componentStates.flatMap((state) =>
```

Comment:
nit: can the `undefined` values be filtered out of `componentStates` before we do this `flatMap`? I think that would be a little more clear

Reply:
Done